src/sdl2/haptic.rs

```diff
@@ -1,3 +1,54 @@
 //! Haptic Functions
 #[allow(unused)]
 use sys::haptic as ll;
+use sys::joystick as sys_joystick;
+
+use HapticSubsystem;
+use common::IntegerOrSdlError;
+use get_error;
+
+impl HapticSubsystem {
+    /// Attempt to open the joystick at number `id` and return it.
+    pub fn open_from_joystick_id(&self, joystick_index: i32) -> Result<Haptic, IntegerOrSdlError> {
+        use common::IntegerOrSdlError::*;
+
+        let haptic = unsafe {
+            let joystick = sys_joystick::SDL_JoystickOpen(joystick_index);
+            ll::SDL_HapticOpenFromJoystick(joystick)
+        };
+
+        if haptic.is_null() {
+            Err(SdlError(get_error()))
+        } else {
+            unsafe { ll::SDL_HapticRumbleInit(haptic) };
+            Ok(Haptic {
+                subsystem: self.clone(),
+                raw: haptic,
+            })
+        }
+    }
+}
+
+#[allow(unused)]
```

Comment:
Why did you allow this unused ? Did you add this at first to avoid unimportant warnings and forgot to delete it later ?